File: VB/TransparentEditorExample/AlphaBlendTextEdit/AlphaBlendTextEdit.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports DevExpress.XtraEditors
Imports TransparentEditorExample

Namespace ZBobb
	Public Class AlphaBlendTextEdit
		Inherits TextEdit
		Shared Sub New()
			RepositoryItemAlphaBlendTextEdit.RegisterAlphaBlendTextEdit()
		End Sub

		Public Sub New()
			MyBase.New()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemAlphaBlendTextEdit.AlphaBlendTextEditName
			End Get
		End Property

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemAlphaBlendTextEdit
			Get
				Return CType(MyBase.Properties, RepositoryItemAlphaBlendTextEdit)
			End Get
		End Property

		Protected Overrides Function CreateMaskBoxInstance() As TextBoxMaskBox
			Return New AlphaBlendTextBox(Me)
		End Function
	End Class
End Namespace

File: CS/TransparentEditorExample/AlphaBlendTextEdit/AlphaBlendTextEdit.cs
using System.ComponentModel;
using DevExpress.XtraEditors;
using TransparentEditorExample;

namespace ZBobb {
    public class AlphaBlendTextEdit :TextEdit {
        static AlphaBlendTextEdit() {
            RepositoryItemAlphaBlendTextEdit.RegisterAlphaBlendTextEdit();
        }

        public AlphaBlendTextEdit() : base() { }

        public override string EditorTypeName {
            get { return RepositoryItemAlphaBlendTextEdit.AlphaBlendTextEditName; }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemAlphaBlendTextEdit Properties {
            get { return (RepositoryItemAlphaBlendTextEdit)base.Properties; }
        }

        protected override TextBoxMaskBox CreateMaskBoxInstance() {
            return new AlphaBlendTextBox(this);
        }
    }
}